sql-cli/tests/test_project.py

```diff
@@ -0,0 +1,42 @@
+import os
+import tempfile
+from pathlib import Path
+from typing import List
+
+from sql_cli.project import BASE_SOURCE_DIR, Project
+
+
+def ls(dir_name: str) -> List[Path]:
+    """
+    Return sorted list of files and directories available in the given directory.
+
+    :param dir_name: Source directory name
+
+    :returns: Sorted list of files and directories within the given directory.
+
+    """
+    return sorted([path.relative_to(dir_name) for path in Path(dir_name).rglob("*")])
+
+
+def test_initialise_project_with_dirname():
+    with tempfile.TemporaryDirectory() as dir_name:
+        assert not os.listdir(dir_name)
```

Comment:
I personally prefer `pathlib` instead of `os` for path operations. I think all of what you are doing can be rewritten with pathlib :)